clientapi/routing/profile.go

```diff
@@ -261,6 +263,12 @@ func updateProfile(
 		return jsonerror.InternalServerError(), err
 	}
 
+	_, domain, err := gomatrixserverlib.SplitID('@', userID)
```

Comment:
This is untrusted, it isn't from a Device. Need to check if the domain is valid.